### PR TITLE
Some basic updates

### PR DESCRIPTION
**Pull request by bjd2385**
There's plenty more here to work on. As always, comments/criticism welcome :)

### CODE REVIEW

File: mimesis/decorators.py [hunk 1]
```diff
@@ -7,23 +9,24 @@
 
 from mimesis import data
 from mimesis.exceptions import UnsupportedLocale
-import mimesis.typing as types
 
+import functools
```
Comment (lk-geimfari) [inline mimesis/decorators.py]: pep8, functools is in standard library.

File: mimesis/decorators.py [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import functools
+from typing import Any, Callable, TypeVar, Generator, Iterable, cast
+U = TypeVar('U', covariant=True)
```
Comment (lk-geimfari) [inline mimesis/decorators.py]: What is mean `U`?

File: mimesis/providers/address.py [hunk 3]
```diff
@@ -84,7 +87,7 @@ def address(self) -> str:
 
         )
 
-    def state(self, abbr: bool = False) -> str:
+    def state(self, abbr: bool =False) -> str:
```
Comment (lk-geimfari) [inline mimesis/providers/address.py]: https://www.python.org/dev/peps/pep-3107/#syntax

File: mimesis/providers/numbers.py [hunk 4]
```diff
@@ -1,14 +1,15 @@
-import array
-import mimesis.typing as types
+from mimesis import typing as types
 
 from mimesis.providers.base import BaseProvider
 
+import array
```
Comment (lk-geimfari) [inline mimesis/providers/numbers.py]: 1. standard library imports
2. related third party imports
3. local application/library specific imports

File: mimesis/decorators.py [hunk 5]
```diff
@@ -37,16 +40,16 @@ def wrapper(*args, **kwargs):
                     'Locale {0} is not supported yet.'.format(locale),
                 )
             result = func(*args, **kwargs)
-            txt = ''.join([alphabet[i] for i in result if i in alphabet])
+            txt = ''.join(cast(Iterable, (alphabet[i] for i in result if i in alphabet)))
```
Comment (bjd2385) [inline mimesis/decorators.py]: I'm having a problem with `join` here, so this isn't yet in a working condition; e.g., I shouldn't have tried to `cast` this value to `Iterable`, it's just meant to show that `join` is complaining about requiring an `Iterable` type. Per `help(''.join)`:

> join(...) method of builtins.str instance
    S.join(iterable) -> str

File: mimesis/decorators.py [hunk 6]
```diff
@@ -56,16 +59,16 @@ def type_to(new_type: types.Callable,
...
+        def wrapper(*args: Any, **kwargs: Any) -> U:
             result = func(*args, **kwargs)
             result = new_type(result)
 
             if check_len and len(result) == 1:
                 return result[0]
             return result
 
-        return wrapper
+        return cast(U, wrapper)
```
Comment (bjd2385) [inline mimesis/decorators.py]: I'm still working out how to annotate these decorators properly :P

File: mimesis/helpers.py [hunk 7]
```diff
@@ -18,10 +20,9 @@ def randints(self, amount: int = 3, a: int = 1, b: int = 100) -> list:
         if not amount:
```
Comment (bjd2385) [inline mimesis/helpers.py]: I feel like we should restrict this further now that I'm looking at it; e.g., the predicate could probably be `amount == 0`, and in an `else` clause `raise ValueError('Amount out of range')` or something.

File: mimesis/providers/numbers.py (hunk 4 above)
Comment (bjd2385) [reply mimesis/providers/numbers.py]: I'll follow that convention from here on out then.

File: mimesis/providers/address.py (hunk 3 above)
Comment (bjd2385) [reply mimesis/providers/address.py]: If that's the convention you'd like to see in this code base, then that's what I'll follow.

File: mimesis/decorators.py (hunk 2 above)
Comment (bjd2385) [reply mimesis/decorators.py]: @lk-geimfari It's a type variable. I've already updated this too in a commit i'll make later with fixes as to your other comments here.

**Comment by lk-geimfari** [comment]
@bjd2385 I suggest pull latest changes from `type-hinting` branch.

**Comment by bjd2385** [comment]
@lk-geimfari Will do. As you can probably tell, I'm still learning about version control and GitHub's workings. Thanks for being so patient with me :)

**Comment by lk-geimfari** [comment]
@bjd2385 Thank you for the contributing!

**Comment by lk-geimfari** [comment]
@bjd2385 Ping.

**Pull request closed without merging**